src/daft-micropartition/src/micropartition.rs

```diff
@@ -918,6 +952,58 @@ fn _get_file_column_names<'a>(
     }
 }
 
+fn _read_delete_files(
+    delete_files: &[&str],
+    io_client: Arc<IOClient>,
+    io_stats: Option<IOStatsRef>,
+    num_parallel_tasks: usize,
+    runtime_handle: Arc<tokio::runtime::Runtime>,
+    schema_infer_options: &ParquetSchemaInferenceOptions,
+) -> DaftResult<HashMap<String, Vec<i64>>> {
+    let columns: Option<&[&str]> = Some(&["file_path", "pos"]);
+
+    let tables = read_parquet_bulk(
+        delete_files,
+        columns,
+        None,
+        None,
+        None,
+        None,
+        io_client,
+        io_stats,
+        num_parallel_tasks,
+        runtime_handle,
+        schema_infer_options,
+        None,
+        None,
+        None,
+    )?;
+
+    let mut delete_map: HashMap<String, Vec<i64>> = HashMap::new();
+
+    for table in tables.iter() {
+        let file_paths = table.get_column("file_path")?.downcast::<Utf8Array>()?;
```

Comment:
Can we add a comment talking about how this `file_path` column is guaranteed by the iceberg spec to match the filepaths that we received?

Also I guess we should be able to skip adding this to the HashMap if it does not exist in `delete_files`? We could do something like:

```
let mut delete_map = {fp: [] for fp in delete_files}
...
```